io/bufio.ts

```diff
@@ -84,14 +80,15 @@ export class BufReader implements Reader {
 
     // Read new data: try a limited number of times.
     for (let i = MAX_CONSECUTIVE_EMPTY_READS; i > 0; i--) {
-      let rr: ReadResult = await this.rd.read(this.buf.subarray(this.w));
-      assert(rr.nread >= 0, "negative read");
-      this.w += rr.nread;
-      if (rr.eof) {
+      const rr = await this.rd.read(this.buf.subarray(this.w));
+      const nread = rr === Deno.EOF ? 0 : rr;
```

Comment:
You wouldn't need this if you moved the `assert(nread >= 0)` *after* the `if (rr ===  EOF) { return}`.